ors-engine/src/main/java/org/heigit/ors/export/ExportRequest.java

```diff
@@ -114,22 +122,52 @@ public ExportResult computeExport(RoutingProfile routingProfile) {
 
         // calculate node coordinates
         for (int from : nodesInBBox) {
-            Coordinate coord = new Coordinate(nodeAccess.getLon(from), nodeAccess.getLat(from));
-            res.addLocation(from, coord);
-
+            Coordinate fromCoords = new Coordinate(nodeAccess.getLon(from), nodeAccess.getLat(from));
+            res.addLocation(from, fromCoords);
             EdgeIterator iter = explorer.setBaseNode(from);
             while (iter.next()) {
                 int to = iter.getAdjNode();
                 if (nodesInBBox.contains(to)) {
-                    double weight = weighting.calcEdgeWeight(iter, false, EdgeIterator.NO_EDGE);
                     Pair<Integer, Integer> p = new Pair<>(from, to);
-                    res.addEdge(p, weight);
+                    Map<String, Object> extra = new HashMap<>();
+                    res.addEdge(p, weighting.calcEdgeWeight(iter, false, NO_TIME));
+                    LOGGER.debug("Edge %d: from %d to %d".formatted(iter.getEdge(), from, to));
+
+                    if (topoJson) {
+                        LineString geo;
+                        if (useRealGeometry) {
+                            geo = iter.fetchWayGeometry(FetchMode.ALL).toLineString(false);
+                        } else {
+                            Coordinate toCoords = new Coordinate(nodeAccess.getLon(to), nodeAccess.getLat(to));
+                            geo = geometryFactory.createLineString(new Coordinate[]{fromCoords, toCoords});
+                        }
+                        if (osmIdsAvailable) {
+                            boolean reverse = iter.getEdgeKey() % 2 == 1;
+                            TopoGeometry topoGeometry = topoGeometries.computeIfAbsent(osmIdGraphStorage.getEdgeValue(iter.getEdge()), x ->
+                                    new TopoGeometry(weighting.getSpeedCalculator().getSpeed(iter, reverse, NO_TIME),
+                                            weighting.getSpeedCalculator().getSpeed(iter, !reverse, NO_TIME))
+                            );
+                            topoGeometry.getArcs().compute(iter.getEdge(), (k, v) -> {
+                                if (v != null) {
+                                    topoGeometry.setBothDirections(true);
+                                    return v;
+                                } else {
+                                    return reverse ? new TopoArc(geo.reverse(), iter.getDistance(), iter.getAdjNode(), iter.getBaseNode()) :
+                                            new TopoArc(geo, iter.getDistance(), iter.getBaseNode(), iter.getAdjNode());
+                                }
+                            });
+                        } else {
+                            extra.put("geometry", geo);
+                        }
+                    }
 
-                    if (debug()) {
-                        Map<String, Object> extra = new HashMap<>();
+                    if (debug) {
```

Comment:
This is an API flag that should be adequately named - and not "debug" which sounds like a service-wide debug flag.

Reply:
done